tests/functional/modules/test_zos_zfs_resize_func.py

```diff
@@ -0,0 +1,551 @@
+            assert result.get('new_free_space') >= result.get('old_free_space')
+            assert result.get('new_size') >= grow_size
+
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=shrink_size,
+                                            space_type=space_type)
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 0
+            assert "shrunk" in result.get('stdout')
+            assert result.get('new_size') <= result.get('old_size')
+            assert result.get('new_free_space') <= result.get('old_free_space')
+            assert result.get('space_type') == space_type.upper()
+            assert result.get('new_size') <= shrink_size
+    finally:
+        clean_up_environment(hosts=hosts, ds_name=ds_name, temp_dir_name=mount_folder)
+
+def test_grow_n_shrink_operations_size_trk(ansible_zos_module):
+    hosts = ansible_zos_module
+    ds_name = get_tmp_ds_name()
+    mount_folder = ""
+    grow_size = 400
+    shrink_size = 200
+    space_type = "trk"
+    try:
+        mount_folder = set_environment(ansible_zos_module=hosts, ds_name=ds_name)
+
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=grow_size,
+                                            space_type=space_type)
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 0
+            assert "grown" in result.get('stdout')
+            assert result.get('new_size') >= grow_size
+            assert result.get('new_size') >= result.get('old_size')
+            assert result.get('new_free_space') >= result.get('old_free_space')
+            assert result.get('space_type') == space_type.upper()
+
+        results = hosts.all.zos_zfs_resize(target=ds_name,
+                                            size=shrink_size,
+                                            space_type=space_type)
+        for result in results.contacted.values():
+            assert result.get('target') == ds_name
+            assert result.get('mount_target') == "/SYSTEM" + mount_folder
+            assert result.get('rc') == 0
+            assert "shrunk" in result.get('stdout')
+            assert result.get('new_size') <= result.get('old_size')
+            assert result.get('new_free_space') <= result.get('old_free_space')
+            assert result.get('new_size') <= shrink_size
+            assert result.get('space_type') == space_type.upper()
+    finally:
+        clean_up_environment(hosts=hosts, ds_name=ds_name, temp_dir_name=mount_folder)
+
+def test_grow_n_shrink_operations_size_cyl(ansible_zos_module):
```

Comment:
These test cases "test_grow_n_shrink_operations_size_cyl" "test_grow_n_shrink_operations_size_m" "test_grow_n_shrink_operations_size_trk" can be parametrized